spock-core/src/main/java/org/spockframework/compiler/ConditionRewriter.java

```diff
@@ -495,16 +495,24 @@ public void visitTupleExpression(TupleExpression expr) {
 
   private Expression record(Expression expr) {
     // replace expr with $spock_valueRecorder.record($spock_valueRecorder.startRecordingValue(recordCount++), <expr>)
-    return AstUtil.createDirectMethodCall(
-        new VariableExpression(SpockNames.VALUE_RECORDER),
-        resources.getAstNodeCache().ValueRecorder_Record,
-        new ArgumentListExpression(
-            AstUtil.createDirectMethodCall(
-                new VariableExpression(SpockNames.VALUE_RECORDER),
-                resources.getAstNodeCache().ValueRecorder_StartRecordingValue,
-                new ArgumentListExpression(new ConstantExpression(recordCount++))
-            ),
-            expr));
+    MethodCallExpression result = AstUtil.createDirectMethodCall(
```

Comment:
Does it make sense to create this, if we discard it in the case of a `CastExpression`? Also doesn't this mess up the `recordCount`?

Reply:
We don't discard it, it is the second argument to the `CastExpression` constructor and thus casts the value returned from the method call. It copies the cast from the recorded expression "around" the record call.